### ISSUE TITLE
Warning: a promise was created in a  handler - When discover=true

### ISSUE THREAD

**Issue by astanciu** [neumino/rethinkdbdash#206]
When `discover: true`, this code:
```
var conf =  {
  host: 'myhost.com',
  port: 28015,
  db: 'the-db',
  discovery: true,
  silent: true
}

var r = require('rethinkdbdash')(conf)
```

When ran in dev mode like so:
`NODE_ENV=development node temp.js`

Produces this warning: 
```
Warning: a promise was created in a  handler but was not returned from it
    at Cursor._next (/db-wrapper/node_modules/rethinkdbdash/lib/cursor.js:54:11)
    at Cursor._each (/db-wrapper/node_modules/rethinkdbdash/lib/cursor.js:251:8)
    at /db-wrapper/node_modules/rethinkdbdash/lib/pool_master.js:304:10
    at processImmediate [as _immediateCallback] (timers.js:368:17)
From previous event:
    at PoolMaster.fetchServers (/db-wrapper/node_modules/rethinkdbdash/lib/pool_master.js:296:11)
    at null._onTimeout (/db-wrapper/node_modules/rethinkdbdash/lib/pool_master.js:73:50)
    at Timer.listOnTimeout (timers.js:92:15)

Warning: a promise was created in a  handler but was not returned from it
    at Cursor._fetch (/db-wrapper/node_modules/rethinkdbdash/lib/cursor.js:116:11)
    at Cursor._push (/db-wrapper/node_modules/rethinkdbdash/lib/cursor.js:137:86)
    at /db-wrapper/node_modules/rethinkdbdash/lib/cursor.js:119:10
    at processImmediate [as _immediateCallback] (timers.js:368:17)
From previous event:
    at Cursor._fetch (/db-wrapper/node_modules/rethinkdbdash/lib/cursor.js:118:6)
    at Cursor._push (/db-wrapper/node_modules/rethinkdbdash/lib/cursor.js:137:86)
    at Connection._processResponse (/db-wrapper/node_modules/rethinkdbdash/lib/connection.js:410:14)
    at Socket.<anonymous> (/db-wrapper/node_modules/rethinkdbdash/lib/connection.js:156:14)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at readableAddChunk (_stream_readable.js:146:16)
    at Socket.Readable.push (_stream_readable.js:110:10)
    at TCP.onread (net.js:523:20)

Warning: a promise was created in a  handler but was not returned from it
    at Cursor._fetch (/db-wrapper/node_modules/rethinkdbdash/lib/cursor.js:116:11)
From previous event:
    at Cursor._fetch (/db-wrapper/node_modules/rethinkdbdash/lib/cursor.js:118:6)
    at Cursor._push (/db-wrapper/node_modules/rethinkdbdash/lib/cursor.js:137:86)
    at Connection._processResponse (/db-wrapper/node_modules/rethinkdbdash/lib/connection.js:410:14)
    at Socket.<anonymous> (/db-wrapper/node_modules/rethinkdbdash/lib/connection.js:156:14)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at readableAddChunk (_stream_readable.js:146:16)
    at Socket.Readable.push (_stream_readable.js:110:10)
    at TCP.onread (net.js:523:20)
```

Without discovery, the warning is not shown. I've seen other similar Issues related to this warning, but thought you should know about this specific instance as well, as it's different than the others.

**Comment by neumino** [comment]
I can't reproduce this with the last version of rethinkdbdash. What version are you using?

**Comment by neumino** [comment]
Never mind, I can reproduce this now.

**Comment by neumino** [comment]
Fixed in 2.2.12

**Comment by astanciu** [comment]
:smile:  Looks good, thanks for the fast turnaround!
